db/log_reader.cc

```diff
@@ -512,6 +528,22 @@ bool FragmentBufferedReader::ReadRecord(Slice* record, std::string* scratch,
         }
         break;
 
+      case kSetCompressionType: {
```

Comment:
Would be good to validate that there's only one such record per WAL file, and that its in the beginning.

Reply:
Added check.